pkg/client/aad.go

```diff
@@ -50,9 +51,18 @@ func (c *aadClientImpl) GetAadToken(ctx context.Context, clientID, clientSecret,
 
 	c.Logger.WithField("scopes", strings.Join(scopes, ",")).Info("requesting new AAD token")
 
-	authResult, err := client.AcquireTokenByCredential(ctx, scopes)
+	authResult, err := retry.DoWithData(func() (confidential.AuthResult, error) {
+		authResult, err := client.AcquireTokenByCredential(ctx, scopes)
+		if err != nil {
+			return confidential.AuthResult{}, err
+		}
+		return authResult, nil
+	}, retry.Context(ctx),
+		retry.Attempts(getAadTokenMaxRetries),
+		retry.MaxDelay(getAadTokenMaxDelay),
+		retry.DelayType(retry.RandomDelay))
```

Comment:
wonder what our convention is around back-off/delay type, not quite sure yet which we should be using for each case